pkgs/os-specific/linux/nvidia-x11/generic.nix

```diff
@@ -59,7 +59,12 @@ let
                    ]
               else [ (fetchurl {
                       url = https://git.archlinux.org/svntogit/packages.git/plain/trunk/kernel_4.10.patch?h=packages/nvidia;  sha256 = "0zhpx3baq2pca2pmz1af5cp2nzjxjx0j9w5xrdy204mnv3v2708z";
-                     }) ];
+                     })
+                     (optional (kernel.features.grsecurity or false) (fetchurl {
```

Comment:
`optional` expands to a singleton list so you'd ideally append it to the `patches` list; as it is, it creates a list-of-lists.